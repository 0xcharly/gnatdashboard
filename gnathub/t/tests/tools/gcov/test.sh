# for idempotence

gprclean -q -Pdefault

# build
gprbuild -q -Pdefault

cd obj

# run the executable
./hello

# compute converage information
gcov hello f 2>&1 > /dev/null

cd ..

# launch gnathub
gnathub -q -Pdefault

# launch the print script
gnathub -q -Pdefault --exec script.py
