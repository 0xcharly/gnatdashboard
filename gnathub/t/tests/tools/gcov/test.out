Second path taken
[hello.adb]
    all_lines = 22 
    code_lines = 19 
    comment_lines = 0 
    eol_comments = 0 
    comment_percentage = 0.00 
    blank_lines = 3 
    statement_complexity = 2.50 
    expression_complexity = 0.00 
    essential_complexity = 2.50 
    cyclomatic_complexity = 2.50 
    max_loop_nesting = 2.00 
    line 4:	coverage = 1
    line 7:	coverage = 1
    line 9:	coverage = 5
    line 10:	coverage = 213
    line 11:	coverage = 209
    line 12:	coverage = 1
    line 14:	coverage = 208
    line 15:	coverage = 4
    line 16:	coverage = 0
    line 19:	coverage = 1
    line 20:	coverage = 1
    line 21:	coverage = 1
    line 22:	coverage = 1
[f.ads]
    all_lines = 5 
    code_lines = 3 
    comment_lines = 0 
    eol_comments = 0 
    comment_percentage = 0.00 
    blank_lines = 2 
[f.adb]
    all_lines = 18 
    code_lines = 11 
    comment_lines = 3 
    eol_comments = 0 
    comment_percentage = 21.42 
    blank_lines = 4 
    statement_complexity = 2.00 
    expression_complexity = 0.00 
    essential_complexity = 1.00 
    cyclomatic_complexity = 2.00 
    max_loop_nesting = 0.00 
    line 9:	coverage = 1
    line 11:	coverage = 1
    line 12:	coverage = 0
    line 14:	coverage = 1
    line 16:	coverage = 1
